src/commands/scanner/run/dfa.ts

```diff
@@ -106,17 +72,13 @@ export default class Dfa extends ScannerRunCommand {
 		// END: Config-overrideable engine flags.
 	};
 
-	protected async validateCommandFlags(): Promise<void> {
+	protected async validateVariantFlags(): Promise<void> {
 		const fh = new FileHandler();
-		// Entries in the projectdir array must be non-glob paths to existing directories.
-		for (const dir of (this.flags.projectdir as string[])) {
-			if (globby.hasMagic(dir)) {
-				throw SfdxError.create('@salesforce/sfdx-scanner', 'run-dfa', 'validations.projectdirCannotBeGlob', []);
-			} else if (!(await fh.exists(dir))) {
-				throw SfdxError.create('@salesforce/sfdx-scanner', 'run-dfa', 'validations.projectdirMustExist', []);
-			} else if (!(await fh.stats(dir)).isDirectory()) {
-				throw SfdxError.create('@salesforce/sfdx-scanner', 'run-dfa', 'validations.projectdirMustBeDir', []);
-			}
+		// The superclass will validate that --projectdir is well-formed,
+		// but doesn't require that the flag actually be present.
+		// So we should make sure it exists here.
+		if (!this.flags.projectdir || (this.flags.projectdir as string[]).length === 0) {
```

Comment:
Since `--projectdir/-p` is now optional (to allow its reuse in both commands), we need to make sure it's present here.